ui/page/staking/overview.go

```diff
@@ -79,16 +89,33 @@ func (pg *Page) ID() string {
 // the page is displayed.
 // Part of the load.Page interface.
 func (pg *Page) OnNavigatedTo() {
-
 	pg.loadPageData() // starts go routines to refresh the display which is just about to be displayed, ok?
 
-	go pg.WL.GetVSPList()
-	// TODO: automatic ticket purchase functionality
-	pg.autoPurchaseEnabled.SetEnabled(false)
+	pg.updateTBToggle()
+}
+
+func (pg *Page) updateTBToggle() {
+	if pg.WL.MultiWallet.TicketBuyerConfigIsSet() {
+		_, walID, _, _ := pg.WL.MultiWallet.GetAutoTicketsBuyerConfig()
+		pg.autoPurchase.SetChecked(pg.WL.MultiWallet.IsAutoTicketsPurchaseActive(walID))
+	} else {
+		pg.autoPurchase.SetChecked(false)
+	}
 }
 
 func (pg *Page) loadPageData() {
-	pg.ticketPrice = dcrutil.Amount(pg.WL.TicketPrice()).String()
+	go func() {
+		ticketPrice, err := pg.WL.MultiWallet.TicketPrice()
```

Comment:
Does this take long to execute? If not, doesn't need to be in a goroutine. If yes, is the UI updated once the price is gotten? Also, is it possible to have all the required data loaded in one goroutine and the page only refreshed after all data is loaded? Can also use multiple goroutines and use a `sync.WaitGroup` to wait for all goroutines to complete before refreshing the page.

Reply:
This has been resolved in https://github.com/planetdecred/godcr/pull/770